### ISSUE TITLE
poudriere jail -c fails to build with non-default MAKEOBJDIRPREFIX

### ISSUE THREAD

**Issue by jbeich** [freebsd/poudriere#368]
[build(7)](https://man.freebsd.org/build/7) documents how to build world with read-only `/usr` or as regular user. If `MAKEOBJDIRPREFIX` is set poudriere is influenced by user environment but not enough to respect it.

``` sh
$ echo $MAKEOBJDIRPREFIX
/home/foo/.cache

$ poudriere jail -cj test-mips -a mips -v head -m svn
[00:00:00] ====>> Cross-building ports for mips on amd64 requires QEMU
[00:00:00] ====>> Creating test-mips fs... done
[00:00:01] ====>> Checking out the sources from svn... done
[00:09:14] ====>> Starting make buildworld with 4 jobs
--- buildworld ---
--- buildworld_prologue ---
--------------------------------------------------------------
>>> World build started on Mon Dec  6 21:43:16 UTC 2015
--------------------------------------------------------------
--- _worldtmp ---

--------------------------------------------------------------
>>> Rebuilding the temporary build tree
--------------------------------------------------------------
rm -rf /usr/obj/mips.mips/poudriere/jails/test-mips/usr/src/tmp
mkdir -p /usr/obj/mips.mips/poudriere/jails/test-mips/usr/src/tmp/lib
mkdir: /usr/obj: No such file or directory
*** [_worldtmp] Error code 1

make[1]: stopped in /poudriere/jails/test-mips/usr/src
1 error

make[1]: stopped in /poudriere/jails/test-mips/usr/src
*** [buildworld] Error code 2

make: stopped in /poudriere/jails/test-mips/usr/src
1 error

make: stopped in /poudriere/jails/test-mips/usr/src
[00:09:14] ====>> Error: Failed to 'make buildworld'
[00:09:14] ====>> Error while creating jail, cleaning up.
[00:09:14] ====>> Removing test-mips jail... done
```

**Comment by bdrewery** [comment]
The `poudriere` command clears the environment. You would need to export it via _poudriere.conf_.

**Comment by jbeich** [comment]
> The poudriere command clears the environment.

git master still fails with `env -i MAKEOBJDIRPREFIX=/path/ ...` but not `env -i ...` which is contrary to your statement.

> You would need to export it via poudriere.conf.

```sh
export MAKEOBJDIRPREFIX=$MAKEOBJDIRPREFIX
```
helps and so does the following patch
```diff
--- a/src/share/poudriere/jail.sh~
+++ b/src/share/poudriere/jail.sh
@@ -352,7 +352,7 @@ build_and_install_world() {
 		msg "Starting make native-xtools with ${PARALLEL_JOBS} jobs"
 		${MAKE_CMD} -C /usr/src native-xtools ${MAKE_JOBS} \
 		    ${MAKEWORLDARGS} || err 1 "Failed to 'make native-xtools'"
-		XDEV_TOOLS=/usr/obj/${TARGET}.${TARGET_ARCH}/nxb-bin
+		XDEV_TOOLS=${MAKEOBJDIRPREFIX:-/usr/obj}/${TARGET}.${TARGET_ARCH}/nxb-bin
 		rm -rf ${JAILMNT}/nxb-bin || err 1 "Failed to remove old native-xtools"
 		mv ${XDEV_TOOLS} ${JAILMNT} || err 1 "Failed to move native-xtools"
 		cat > ${JAILMNT}/etc/make.nxb.conf <<- EOF
```
Is this expected?

**Comment by bdrewery** [comment]
We use `env -i` internally to clear the environment: https://github.com/freebsd/poudriere/blob/master/src/bin/poudriere#L141

**Comment by bdrewery** [comment]
The `XDEV_TOOLS` code was fixed in 439fde2c7bff59a970a73d289996493f9e7eb20e.
